pkg/appsec/config/config.go

```diff
@@ -0,0 +1,68 @@
+package config
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/pkg/errors"
+)
+
+// AgentConfig represents the agent configuration API.
+type AgentConfig interface {
```

Comment:
I recommend removing this. There are already mechanisms in place to facilitate testing the global configuration that you can use. No need to add extra layers. Furthermore, this isn't used now.

Reply:
Done: replaced by the existing interface type `(pkg/config).Config`